polars/dataframe.ts

```diff
@@ -702,7 +702,7 @@ export interface DataFrame
    * @param options.on - Name(s) of the join columns in both DataFrames.
    * @param options.how - Join strategy
    * @param options.suffix - Suffix to append to columns with a duplicate name.
-   * @see {@link JoinOptions}
```

Comment:
i think this should link to `JoinOptions`. the `JoinBaseOptions` is just a base type used to prevent duplication when creating the method overrides. 

The `JoinOptions` represent the full options available for `join`. 

Alternatively this could be expressed as `join(df: DataFrame, options: JoinOptions)` and remove all of the overrides, but the overrides provide better type hints than one generic `options` object.  such as not being able to specify both `on`, and `leftOn + rightOn`.

Reply:
Thanks for the explanation! I'll fix this in a new commit later 😀

Reply:
I reverted the change, I think I mostly got confused because the link attribute does not directly work, see:
![image](https://user-images.githubusercontent.com/43115095/209481289-197f3bcd-360a-4145-beb6-71b2861dc10d.png)